Docs.rs build is broken

**Describe the bug**
I broke docs.rs build by merging https://github.com/apache/arrow-rs/pull/1349 without testing it first


**To Reproduce**
I have tested / reproduced by running the builds from this branch:  https://github.com/apache/arrow-rs/compare/master...alamb:alamb/test_crates_publish

This is what happening 

https://docs.rs/crate/parquet/0.18.0 

Check logs: https://docs.rs/crate/parquet/0.18.0/builds/557342

```
[INFO] [stderr] WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
[INFO] [stdout] c7db694a1e0ec0a7df60f4774a81f3d15861a69233e9177971982b83d9da03dd
[INFO] running `Command { std: "docker" "start" "-a" "c7db694a1e0ec0a7df60f4774a81f3d15861a69233e9177971982b83d9da03dd", kill_on_drop: false }`
[INFO] [stderr]  Documenting parquet v0.18.0 (/opt/rustwide/workdir)
[INFO] [stderr] error[E0658]: `#[doc(cfg)]` is experimental
[INFO] [stderr]    --> src/arrow/mod.rs:124:21
[INFO] [stderr]     |
[INFO] [stderr] 124 | #[cfg_attr(doc_cfg, doc(cfg(feature = "async")))]
[INFO] [stderr]     |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[INFO] [stderr]     |
[INFO] [stderr]     = note: see issue #43781 <https://github.com/rust-lang/rust/issues/43781> for more information
[INFO] [stderr]     = help: add `#![feature(doc_cfg)]` to the crate attributes to enable
[INFO] [stderr] 
[INFO] [stderr] error: Compilation failed, aborting rustdoc
[INFO] [stderr] 
[INFO] [stderr] For more information about this error, try `rustc --explain E0658`.
[INFO] [stderr] error: could not document `parquet`
[INFO] [stderr] 
[INFO] [stderr] Caused by:
[INFO] [stderr]   process didn't exit successfully: `rustdoc --edition=2021 --crate-type lib --crate-name parquet src/lib.rs --target x86_64-unknown-linux-gnu -o /opt/rustwide/target/x86_64-unknown-linux-gnu/doc --cfg 'feature="arrow"' --cfg 'feature="async"' --cfg 'feature="base64"' --cfg 'feature="brotli"' --cfg 'feature="clap"' --cfg 'feature="cli"' --cfg 'feature="default"' --cfg 'feature="experimental"' --cfg 'feature="flate2"' --cfg 'feature="futures"' --cfg 'feature="lz4"' --cfg 'feature="serde_json"' --cfg 'feature="snap"' --cfg 'feature="test_common"' --cfg 'feature="tokio"' --cfg 'feature="zstd"' --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat -C metadata=78f9c9e2e62425b8 -L 
...
```


**Expected behavior**
docs.rs build should succeed 

**Additional context**

YOLO in this case was not a good idea: https://github.com/apache/arrow-rs/pull/1349#issuecomment-1125086652

--- a/parquet/Cargo.toml
+++ b/parquet/Cargo.toml
@@ -63,7 +63,6 @@ arrow = { path = "../arrow", version = "13.0.0", default-features = false, featu
 
 [package.metadata.docs.rs]
 all-features = true
-rustdoc-args = ["--cfg", "doc_cfg"]
 
 [features]
 default = ["arrow", "snap", "brotli", "flate2", "lz4", "zstd", "base64"]

--- a/parquet/src/arrow/mod.rs
+++ b/parquet/src/arrow/mod.rs
@@ -121,7 +121,6 @@ pub mod arrow_writer;
 mod bit_util;
 
 #[cfg(feature = "async")]
-#[cfg_attr(doc_cfg, doc(cfg(feature = "async")))]
 pub mod async_reader;
 
 experimental_mod!(converter);

